controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/billing/BillingApiHandler.java

```diff
@@ -101,10 +105,28 @@ private HttpResponse handleGET(HttpRequest request, Path path, String userId) {
         if (path.matches("/billing/v1/tenant/{tenant}/billing")) return getBilling(path.get("tenant"), request.getProperty("until"));
         if (path.matches("/billing/v1/tenant/{tenant}/plan")) return getPlan(path.get("tenant"));
         if (path.matches("/billing/v1/billing")) return getBillingAllTenants(request.getProperty("until"));
+        if (path.matches("/billing/v1/invoice/export")) return getAllInvoices();
         if (path.matches("/billing/v1/invoice/tenant/{tenant}/line-item")) return getLineItems(path.get("tenant"));
         return ErrorResponse.notFoundError("Nothing at " + path);
     }
 
+    private HttpResponse getAllInvoices() {
+        var invoices = billingController.getInvoices();
+        var headers = new String[]{ "ID", "Tenant", "From", "To", "CpuHours", "MemoryHours", "DiskHours", "Cpu", "Memory", "Disk" };
+        var rows = invoices.stream()
+                .map(invoice -> {
+                    return new Object[] {
+                            invoice.id().value(), invoice.tenant().value(),
+                            invoice.getStartTime().format(DateTimeFormatter.ISO_LOCAL_DATE),
+                            invoice.getEndTime().format(DateTimeFormatter.ISO_LOCAL_DATE),
+                            invoice.sumCpuHours(), invoice.sumMemoryHours(), invoice.sumDiskHours(),
+                            invoice.sumCpuCost(), invoice.sumMemoryCost(), invoice.sumDiskCost()
+                    };
```

Comment:
What about total cost/additional line items?

Reply:
Not supported in the downstream process, yet.